engine-precompiles/src/native.rs

```diff
@@ -243,6 +252,27 @@ fn validate_amount(amount: U256) -> Result<(), ExitError> {
     Ok(())
 }
 
+struct Recipient {
+    receiver_account_id: AccountId,
+    message: Option<String>,
+}
+
+fn parse_recipient(recipient: &[u8]) -> Result<Recipient, ExitError> {
```

Comment:
Tiny nitpick. I would add some unit tests for this function. I believe It would be trivial.